docs/howto/features/auth-provider.md

```diff
@@ -0,0 +1,75 @@
+# Providing credentials for externally managed services
+
+Sometimes, our communities want to run *their own* [external JupyterHub service](https://jupyterhub.readthedocs.io/en/stable/reference/services.html#externally-managed-services),
+often just to use JupyterHub as an OAuth2 provider for user logins. This
+allows them to have the same shared users between the hub and their 
+service.
```

Comment:
I think we should let the title and the example focus solely on registering a jupyterhub service / oauth2 application, and avoid the broader "providing credentials for externally managed services" which should be done differently for all (?) other cases.

If they want to register a service to acquire credentials to interact with the JupyterHub REST API in order to stop user servers or similar, it would not involve `oauth_client_id` or `oauth_redirect_url`, it would instead only involve `api_token` and configuring `load_roles` defining more fine grained permissions to do that.

That makes me think we should scope the docs tighter that this is an example specifically about providing a credentials to use JupyterHub as an identity provider - where we register an OAuth2 application.

Reply:
Yeah, I had originally written this out as 'how to make JupyterHub an auth provider' (hence the file name!), but then tried to broaden it. But as you say, in most other service cases you only need api_token - I hadn't considered that. So I'll revert this back to the previous framing.